### PR TITLE
[Bugfix] scalar constant

### PR DESCRIPTION
**Pull request by xuhongyao**

### CODE REVIEW

File: lib/target/generic_cpp/generic_cxx_codegen.cc [hunk 1]
```diff
@@ -786,7 +786,7 @@ void GenericCXXCodeGen::RunOnConstant(Constant& constant, bool decl) {
   auto& type = constant.GetResultType();
   if (decl) {
     CXXValue value(constant.GetName(), TensorTypeToCXXType(type, true));
-    if (constant.IsScalarOne()) {
+    if (type.GetTotalNumOfElements() == 1) {
```
Comment (weimingzha0) [inline lib/target/generic_cpp/generic_cxx_codegen.cc:789]: Try "type.IsScalar()" ?

File: lib/parser/onnx/onnx_parser.cc [hunk 2]
```diff
@@ -396,7 +396,7 @@ Tensor<T> ONNXParser::ProcessTensor(const onnx::TensorProto& tensor_proto) {
     LOG(ERROR) << "Unsupported external data storage.";
   }
 
-  if (shape.empty() && v.size() > 1) {
+  if (shape.empty() && v.size() >= 1) {
```
Comment (weimingzha0) [inline lib/parser/onnx/onnx_parser.cc:399]: This will make all scalars to be "[1]" shaped tensor.

Comment (weimingzha0) [reply lib/parser/onnx/onnx_parser.cc:399]: I'm afraid this will break other cases

**Comment by weimingzha0** [comment]
there is a typo in commit message: "scale" => "scalar"

File: lib/parser/onnx/onnx_parser.cc (hunk 2 above)
Comment (xuhongyao) [reply lib/parser/onnx/onnx_parser.cc:399]: It is possible to change some input constant parameters from scalar to vector, such as the index of gather.Before scalar processing logic may have to be changed to vector dimension 1

**Pull request closed without merging**